src/relation_handlers.py

```diff
@@ -275,7 +275,24 @@ def _req_already_treated(self, request_id):
         :returns: Whether request is already handled
         :rtype: bool
         """
-        return request_id in self._stored.processed
+        status = relation.data[req_unit]
+        client_unit_name = status.get("unit-name", req_unit.name)
+        response_key = "broker-rsp-" + client_unit_name.replace("/", "-")
+        if not status.get(response_key):
+            return False
+
+        if isinstance(status[response_key], str):
+            try:
+                data = json.loads(status[response_key])
+            except (TypeError, json.decoder.JSONDecodeError):
+                logging.debug("Not ableto decode broker response {req_unit}")
```

Comment:
```suggestion
                logging.debug(f"Not ableto decode broker response {req_unit}")
```
Is the f-string missing ?

Reply:
Done